jans-cedarling/cedarling/src/common/policy_store.rs

```diff
@@ -5,155 +5,430 @@
+    pub person_id: String,
+
+    /// An optional claim used to create a role for the token.
+    pub role_mapping: Option<String>,
+}
+
+#[derive(Debug, Copy, Clone)]
+#[allow(dead_code)]
+pub enum TokenKind {
+    /// Access token used for granting access to resources.
+    Access,
+
+    /// ID token used for authentication.
+    Id,
+
+    /// Userinfo token containing user-specific information.
+    Userinfo,
+
+    /// Transaction token used for tracking transactions.
+    Transaction,
+}
+
+/// Enum representing the different kinds of tokens used by Cedarling.
+impl fmt::Display for TokenKind {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let kind_str = match self {
+            TokenKind::Access => "access",
+            TokenKind::Id => "id",
+            TokenKind::Userinfo => "userinfo",
+            TokenKind::Transaction => "transaction",
+        };
+        write!(f, "{}", kind_str)
     }
+}
 
-    /// A custom deserializer for Cedar's policy set.
-    //
-    // is used to deserialize field `policies` in `models::policy_store::PolicyStore`
-    pub(crate) fn parse<'de, D>(deserializer: D) -> Result<cedar_policy::PolicySet, D::Error>
+impl<'de> Deserialize<'de> for TokenKind {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
     where
         D: serde::Deserializer<'de>,
     {
-        let policies =
-            <HashMap<String, RawPolicy> as serde::Deserialize>::deserialize(deserializer)?;
-
-        let policy_vec = policies
-            .into_iter()
-            .map(|(id, policy_raw)| {
-                let policy = parse_policy::<D>(&id, policy_raw).map_err(|err| {
-                    serde::de::Error::custom(format!(
-                        "unable to decode policy with id: {id}, error: {err}"
-                    ))
-                })?;
-                Ok(policy)
-            })
-            .collect::<Result<Vec<cedar_policy::Policy>, _>>()?;
-
-        cedar_policy::PolicySet::from_policies(policy_vec).map_err(|err| {
-            serde::de::Error::custom(format!("{}: {err}", ParsePolicySetMessage::CreatePolicySet))
-        })
+        struct TokenKindVisitor;
+
+        impl<'de> Visitor<'de> for TokenKindVisitor {
+            type Value = TokenKind;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str("a valid token kind string")
+            }
+
+            fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                match value.to_lowercase().as_str() {
+                    "id" => Ok(TokenKind::Id),
+                    "userinfo" => Ok(TokenKind::Userinfo),
+                    "access" => Ok(TokenKind::Access),
+                    "transaction" => Ok(TokenKind::Transaction),
+                    _ => Err(serde::de::Error::unknown_variant(
```

Comment:
Can we cover in unit tests this error case?

Reply:
test added here: [f9f67d4](https://github.com/JanssenProject/jans/pull/9910/commits/f9f67d4439923145714592b58f319d2d89eecff7)